snap-build/snapcraft-snap-build.yaml

```diff
@@ -196,52 +279,227 @@ parts:
+        for f in `find tmp/FIRMWARE/mediatek/ -type f -name "*.dtb*"`; do
+          filepath="${f%/*}"
+          filename="${f##*/}"
+          cp fdt.its "$filepath"/fdt.its
+          sed -i 's/FDTBIN/'"$filename"'/g' "$filepath"/fdt.its
+          mkimage -F -k ${CRAFT_STAGE}/keys/ -f "$filepath"/fdt.its "$filepath"/"$filename"
+          rm "$filepath"/fdt.its
+        done
+      fi
+
+      mkfs.vfat -n "firmware" -S ${SECTOR_SIZE} -C firmware.vfat ${FW_IMG_SIZE}
+      mcopy -i firmware.vfat -s tmp/FIRMWARE ::
+      cp firmware.vfat "${CRAFT_PART_INSTALL}"
+    organize:
+      '*': usr/lib/boot-firmware/
+
+  sign-boot-firmware:
+    after:
+      - boot-firmware
+      - optee-os
+      - test-keys
+      - trusted-firmware-a
+      - u-boot-mtk
+    plugin: nil
+    # MediaTek tools for secure boot: Repositories that require NDA contract to access
+    source-type: git
+    source: git@gitlab.com:mediatek/aiot/nda/mtk-secure-boot-tools.git
+    #source-branch: python3
+    source-commit: 50dbe8868df51adfc197a1395a202583e2ddbd2d
+    source-depth: 1
+    build-environment:
+      - MEDIA: emmc
+      - BL2: ${CRAFT_STAGE}/bl2.bin
+      - BL31: ${CRAFT_STAGE}/bl31.bin
+      - BL32: ${CRAFT_STAGE}/tee.bin
+      - BL33: ${CRAFT_STAGE}/u-boot.bin
+      - PLATFORM: mt8395
+      - SOC: MT8195
+      - SIGN_EN: 0
+      - SIGN_ARGS: sha256
     override-build: |
-      make -j$(nproc) V=1 E=0 PLAT=${PLATFORM} SPD=opteed PLAT_AB_BOOT_ENABLE=1 PLAT_HW_CRYPTO=${PLAT_HW_CRYPTO} \
-        BUILD_BASE=${CRAFT_PART_BUILD}/build \
-        OPENSSL_DIR=/usr/lib/${CRAFT_ARCH_TRIPLET} \
-        LIBDRAM=${LIBDRAM} LIBBASE=${LIBBASE} \
-        BL32=${BL32} BL33=${BL33} NEED_BL32=yes NEED_BL33=yes CFLAGS+=-DNEED_BL32 \
-        bl31
-      make -j$(nproc) V=1 E=0 PLAT=${PLATFORM} SPD=opteed PLAT_AB_BOOT_ENABLE=1 PLAT_HW_CRYPTO=${PLAT_HW_CRYPTO} \
-        BUILD_BASE=${CRAFT_PART_BUILD}/build \
-        OPENSSL_DIR=/usr/lib/${CRAFT_ARCH_TRIPLET} \
-        LIBDRAM=${LIBDRAM} LIBBASE=${LIBBASE} \
-        BL32=${BL32} BL33=${BL33} NEED_BL32=yes NEED_BL33=yes CFLAGS+=-DNEED_BL32 \
-        bl2
-      cp build/${PLATFORM}/release/bl2.bin bl2.img.tmp
-      truncate -s%4 bl2.img.tmp
-      ${CRAFT_STAGE}/mkimage -T mtk_image -a 0x201000 -e 0x201000 -n "media=${MEDIA};arm64=1" -d bl2.img.tmp bl2.img
+      source ${CRAFT_PROJECT_DIR}/configs/snap.cfg
+      if [ "$CONFIG_SECURE_BOOT" == "y" ]; then
+        ### Create lk.sign for lk.bin
+        cat ${CRAFT_STAGE}/usr/lib/boot-firmware/lk.bin | openssl dgst -binary -${SIGN_ARGS} > lk_digest
+        openssl pkeyutl -sign -inkey ${CRAFT_STAGE}/keys/da.pem -in lk_digest -out lk.sign \
+          -pkeyopt digest:${SIGN_ARGS} -pkeyopt rsa_padding_mode:pss -pkeyopt rsa_pss_saltlen:32
+        cp lk.sign "${CRAFT_PART_INSTALL}"
+        ### Sign bl2.img and create auth_sv5.auth and keyhash
+        cp ${BL2} bl2.img
+        make -C configs PLATFORM=${PLATFORM} SOC=${SOC} \
```

Comment:
This make command will run a elf binary `/root/parts/sign-boot-firmware/build/configs//../sign-image_v2/der_extractor/der_extractor`.

However, `der_extractor` is a x86-64 ELF, and it can not be run on a arm64 system. So either we need to remove `arm64` in the `architectures.build-on` in this yaml file, or we need make `der_extractor` available on arm64 architecture.